service/history/visibilityQueueTaskExecutor.go

```diff
@@ -407,19 +400,20 @@ func (t *visibilityQueueTaskExecutor) recordCloseExecution(
 	searchAttributes *commonpb.SearchAttributes,
 ) error {
 
-	// Record closing in visibility store
-	retentionSeconds := int64(0)
-	namespace := defaultNamespace
-	recordWorkflowClose := true
-
 	namespaceEntry, err := t.shard.GetNamespaceCache().GetNamespaceByID(namespaceID)
-	if err != nil && !isWorkflowNotExistError(err) {
-		return err
+	if err != nil {
+		if _, notFound := err.(*serviceerror.NotFound); !notFound {
+			return err
+		}
 	}
 
-	if err == nil {
-		// retention in namespace config is in days, convert to seconds
-		retentionSeconds = int64(timestamp.DurationFromDays(namespaceEntry.GetRetentionDays(workflowID)).Seconds())
+	var retention *time.Duration
+	namespace := defaultNamespace
+	recordWorkflowClose := true
```

Comment:
should the sampling logic be handled by the visibility sampling store?

Reply:
I was also surprised to see it there but as soon as I am going to replace it with throttling soon, will just leave it there for now.